src/main/java/com/gachigang/ontherun/service/ProductService.java

```diff
@@ -0,0 +1,53 @@
+package com.gachigang.ontherun.service;
+
+import com.gachigang.ontherun.persistence.entity.Category;
+import com.gachigang.ontherun.persistence.entity.Department;
+import com.gachigang.ontherun.persistence.entity.Product;
+import com.gachigang.ontherun.persistence.repository.CategoryRepository;
+import com.gachigang.ontherun.persistence.repository.DepartmentRepository;
+import com.gachigang.ontherun.persistence.repository.ProductRepository;
+import jakarta.persistence.EntityNotFoundException;
+import lombok.AllArgsConstructor;
+import org.springframework.data.domain.Pageable;
+import org.springframework.stereotype.Service;
+
+import java.util.List;
+
+@Service
+@AllArgsConstructor
+public class ProductService {
+
+    private ProductRepository productRepository;
+    private CategoryRepository categoryRepository;
+    private DepartmentRepository departmentRepository;
+
+    public List<Product> getAllByCategory(Long categoryId, Pageable pageable){
+        return productRepository.findAllByCategoryIdAndActiveIsTrue(categoryId, pageable);
+    }
+
+    public List<Product> getAllByDepartment(Long departmentId, Pageable pageable){
+        return productRepository.findAllByDepartmentId(departmentId, pageable);
+    }
+
+    public List<Product> getAllNotActiveProduct(Pageable pageable){
```

Comment:
Delete Pageable pageable, because we get rid of the controller and update the method in the repository